New ui responsive issues

juzen2003 opened the pull request:
# Update
- Update css to have metadata selector modal responsive to window resizing. 
- Item in progress:
  - Make metadata dialog modal responsive.
- Note: in chrome, the minimum height we can shrink to is 296px.

juzen2003 committed:
- This is a commit before switching branches
- In opus.css, select #op-metadata-selector .modal-dialog and set
  max-width to 1100px, this will make metadata selector modal shrink with
  screen width properly.

juzen2003 committed:
- This is a commit before switching branches.
- In browse.js, in updateSliderHandle, add console.log for debugging, will
  have to clean them up later.

juzen2003 committed:
-This is a commit before switching branches

juzen2003 committed:
- In opus.css, add all css related values to make metadata selector modal
  responsive (line 469-503).
- In browse.js, remove all unwanted console.log in updateSliderHandle.

juzen2003 committed:
- In opus.js, call adjustMetadataSelectorMenuPS and adjustSelectedMetadataPS
  in window resize event handler function. This will make sure scrollbar
  in metadata selector is updated when screen is resized.

juzen2003 committed:
-Merge remote-tracking branch 'upstream/new_ui' into new_ui_responsive_issues

juzen2003 committed:
- In browse.js, (at line 930) set .modal-dialog css to top:0, left: 0
  when the modal is open. This will make sure the modal is always at the
  original position when open. It will not be open at the previous closed
  location.

juzen2003 committed:
- In base.html, add modal-dialog-centered to op-metadata-selector modal,
  this will make sure the modal stays at center when open.
- In opus.css, set max-height: calc(75vh - 150px) for .op-all-metadata-column
  and .op-selected-metadata-column, this will make sure buttons not be
  covered when window (height) is very short.

juzen2003 committed:
Merge remote-tracking branch 'upstream/new_ui' into new_ui_responsive_issues

juzen2003 committed:
- In opus.js, call adjustBrowseDialogPS in window resize handler function.
  This will make sure ps in metadata dialog gets updated when resizing.
- In browse.js, in showMetadataDetailModal function, set
  $("#galleryView .modal-dialog").css({top: 0, left: 0}) before opening
  the modal. This will make sure draggable modal is always at the center
  when re-open.
- In opus.css, updated css to make metadata display dialog responsive when
  screen is resized (line 587-629, line 656-662).

juzen2003 committed:
- In opus.css, update img.preview to have max-height: 70vh. This will
  allow us to have bigger image and modal window when resized.

juzen2003 committed:
- In opus.css, add required css to make search tab responsive to window
  resizing (line 141-181, 968-971, 1019-1021).
- In browse.js, modify adjustMetadataSelectorMenuPS, adjustSelectedMetadataPS,
  and adjustBrowseDialogPS to update perfect scrollbar only when perfect
  scrollbar instances are available.

juzen2003 committed:
- In opus.css, modify css to make search page responsive to window resizing
  (line 60-62, 150-192).

juzen2003 committed:
- In opus.css, update line 156-173 to make search page responsive.
- In opus.css, update line 1016, and 947-950. Change padding from % to
  actual px, this will avoid weird behavior of space shrinking when window
  width increases. For border-box, if we set padding to %, it will have
  this behavior, so we change it back to px.

juzen2003 committed:
- In opus.css, update comment at line 157.

juzen2003 committed:
- In base.html, modify navbar-toggler to move "Recent Announcements",
  "Questions/Feedback", and "Help" into hamburger when screen is narrow.
  Rest of tabs options will stay in navbar when screen is narrow.
- In opus.css, update line 178-204 to make reset buttons stay in one line,
  and search menu shrink (won't get cut off) when screen width is about
  600px.

juzen2003 committed:
Merge remote-tracking branch 'upstream/new_ui' into new_ui_responsive_issues

juzen2003 committed:
- Copy over the correct base.html from seti repo, new_ui branch.

juzen2003 committed:
- In opus.css, update line 183-186, this will make search menu shrink more
  smoothly.

juzen2003 committed:
- In base.html, update to have div with .collapse .navbar-collapse wrap
  "Recent Announcements", "Questions/Feedback", and "Help" so that those
  three items will be wrapped into a hamburger when window is narrow.
- In base.html, remove class .collapse .navbar-collapse from
  .browse-nav-container in browse page, so that the 2nd navbar will not
  disappear when part of the 1st navbar is wrapped into hamburger.
- In opus.css, update line 41-53. This will make the 1st navbar looks
  better when hamburger shows up, and also make items in hamburger display
  in one line.
- In opus.css, update line 851-908 to make the 2nd navbar responsive to
  window resizing. When window is narrow, the slider will be moved to the
  2nd line and take up all the width, and it will shrink proportionally
  when screen is getting narrow.

juzen2003 committed:
- In base.html, removed commented out element in line 102.
- In opus.css, clean up comments at line 899.

juzen2003 committed:
- In opus.css, update line 1206-1262 to make detail page responsive to
  window resizing.
- In opus.css, update line 899-894. This will make the slider to take up
  75% of the window width so that slider bar will not be cut off by the
  the screen when it moves to the end.
- In detail.html, add .op-detail-content in .row div. This is for adding
  flex-wrap: wrap-reverse to this .row (flexbox container) in css file
  so that we can move detail image on top of the detail left pane when
  screen is narrow.
- In detail.js, update adjustDetailHeight function to account for detail
  image height when the detail image is moved on top of the left pane.
  That way the container for perfect scrollbar will have the correct height.

juzen2003 committed:
- In opus.css, update line 1236-1240. This is to make sure detail image
  will stay close to the left pane when screen is getting very wide.

juzen2003 committed:
- In base.html, add #op-cart-download-panel as a slide-over panel to display
  download options info in cart page when screen is narrow. Also add a
  nav-link #op-slide-left-pane to open the panel when screen is narrow
  (line 180-184, 226-239)
- In base.html, remove .collapse .navbar-collapse in cart tab .browse-nav-container
  so that the 2nd navbar will not be hidden when hamburger in 1st navbar
  is invoked (line 159).
- In cart.js, add displayCartLeftPane function to move dom elements in
  #op-download-options-container between "#op-cart-download-panel .op-card-contents"
  and ".cart_details" when screen width is narrow (line 111-124).
- In cart.js, add event handlers to display slide-over panel for download
  options (line 95-109).
- In cart.js, modify adjustProductInfoHeight to properly calculate the
  height of download options container so that ps can be updated properly
  (line 175-180).
- In opus.js, add displayCartLeftPaneDB in resize handler function. This
  is to check if we should move dom elements in #op-download-options-container
  between original left pane and slide-over panel. (line 563, 575).
- In opus.js, rename some functions (line 560-562, 572-574).
- In opus.css, make cart tab responsive to window resizing (line 919-997).
- In opus.css, modify selectors to be more specific to the browse tab
  (related to responiveness, line 851-889).
- In browse.js, fix the metadata-dialog dragging issue (line 270).

juzen2003 committed:
- In base.html, removed unwanted comments (line 226-236).

juzen2003 committed:
- In base.html, put slider into a .op-slider-nav ul (line 114). This is
  for easier styling when slider is moving to the 2nd line.
- In opus.css, update to make the 3 options in 2nd navbar stay the same
  without shrinking and growing when the slider is moved to the 2nd line
  (line 879-909).

rfrenchseti commented on opus/application/static_media/js/browse.js:
```diff
@@ -267,7 +267,7 @@ var o_browse = {
         $(".modal-dialog").draggable({
             handle: ".modal-content",
-            cancel: ".metadata",
+            cancel: ".contents",   // Change from ".metadata" to ".contents"
```
You don't need to include the comment here - I was just telling you which line had to change

rfrenchseti reviewed (changes requested):
A few things it would be nice to fix before deployment:

- General:
  - The message that says the browser "works best with" should really say "requires" since we don't even let the user use OPUS below that size.

- Cart:
  - When the browser is fairly small (but still larger than 600x200) and you refresh or switch to the Cart tab, the dataimages.json API gets a call for a *negative* limit.
  - The "Empty Cart" button has moved down further than it was. It should be aligned with the top row of the secondary nav items (View Table, etc.) and stay up there even if the nav bar goes to two lines.
  - When the secondary nav bar becomes two lines, there is a big vertical gap between the first and second lines, much larger than between the second line and the sort order. This gap should be much smaller.
  - The "Download Options" button should have a "click here" mouse pointer (like the other nav bar options) and the text should not be selectable. Right now you can drag-select the text in the button.
  - When the browser gets very narrow, and the secondary nav bar breaks into two lines again, the "Download Options" button stays to the right of all of them, giving you weird things like "Data Archive   URL Archive   <- Download Options". The Download Options should always be on the left.
  - I think I'd rather see the Download Options be below the sort order to better separate it from the nav bar. This would fix the previous issue as well.
  - Also I think maybe it would be nice to make Download Options actually look like a button, or at least to something to bring attention to it.

- Detail:
  The responsiveness isn't working well for some image sizes. For example, look at co-vims-v1484510044_ir. The image and text below it will be partially cut off before it switches to being on top of the left column

A few things that could use tuning but we can do after deployment:

- General:
  - When the browser is narrow enough that the primary nav bar has a hamburger, there is slightly different spacing between the remaining nav items than there was without a hamburger.
  - When the browser is narrow enough that the primary nav bar has a hamburger, click on the hamburger, then Help. The (?) icon next to Help will not have any margin to its left.

- Search:
  - As the browser narrows and the left pane starts to shrink, the horizontal lines under each category run into the vertical divider instead of maintaining some kind of margin. They also run under the PS.
  - In the widgets, the vertical divider on the right and the > and X end up in a weird place if the widget shrinks such that they can't fit on the same line as the widget title.

- Browse:
  - Make the slider fill the available space when the browser is wide.

- Cart:
  - When the browser hits 700px width, the font size of the secondary nav bar changes slightly

- Detail
  - When the browser is very small, the image takes up the whole space and you can no longer see the actual metadata. The image should shrink further in this case.

juzen2003 committed:
- In about.html, update the message about supporting browser size
  (line 350).
- In base.html, make .op-slide-left-pane a button and move it below
  .sort-order-container (line 205-213).
- In browse.js, update calculateGalleryWidth function, change line 1739
  to width <= 0 to avoid wrong dataimages api call (line 1739).
- In detail.js, add suppressScrollX: true to detailPageScrollbar to avoid
  x-scrolling (line 113).
- In opus.js, update the modal message about supporting browser size
  (line 870).
- In opus.css, update values base on comments in #705 (items to be fixed
  before deployment) (line 955-1017, 1336-1391).

juzen2003 committed:
- In base.html, clean up commented out elements.

juzen2003 committed:
- In opus.css, clean up unwanted (commented out) selectors.

juzen2003 committed:
- In opus.css, update values in line 871-878, 915 to reduce vertical spaces
  in the 2nd navbar when there are two lines in 2nd navbar (browse tab).

juzen2003 committed:
- In opus.css, fix the blinking issue when switching between metadata
  dialog (line 680).

juzen2003 committed:
- In opus.css, make sure "x" button in metadata dialog is not cut off
  (line 331-334).

juzen2003 committed:
- In opus.css,  Make #galleryView modal right contents shrink vertically.
  95% for the metadata info, 5% for the bottom icons. (line 692, 739).
